### ISSUE TITLE
Crashlytics plugin - keys don't work

### ISSUE THREAD

**Issue by neiljaywarner** [flutter/flutter#31371]
I tried fabric.io dashboard (an app with the Androidmanifest and such where crashes are reported) and I tried a firebase crashlytics bug

I tried keys regular places and, just in case, right before sent, string, boolean, int, etc
```
void main() {
  Crashlytics.instance.enableInDevMode = true;
  Crashlytics.instance.setUserIdentifier("pretend guid");


  // Pass all uncaught errors to Crashlytics.
  FlutterError.onError = (FlutterErrorDetails details) {
    Crashlytics.instance.setString("Key_within_on_error", "test535");
    Crashlytics.instance.onError(details);
  };
  runApp(MyApp());
}
```
Important note: Crashlytics.instance.log(message) *does* work and the logs do show up, and userid does work.

**Comment by flenairMacif** [comment]
I confirm, _prepareKeys() doesn't seem to work.

**Comment by behnam-lum** [comment]
Same here, I'm using the latest version and keys still don't work!

**Comment by kroikie** [comment]
I'll take a look, thanks for the report.

**Comment by kroikie** [comment]
@All I've had a look and I do see the keys coming through in the Firebase console.

Clarification: Are you looking for the keys in the Firebase console or in the Fabric dashboard?

**Comment by behnam-lum** [comment]
I'm looking at Firebase console. Tried it again with different types of keys but still, I don't see them there. Also tried different kinds of dummy errors but none of them had the keys when they showed up on the console.

**Comment by kroikie** [comment]
I'm unable to reproduce this, keys show up for me as expected. Have you tested with the example app in the plugin?

**Comment by sperochon** [comment]
The same for me but only on iOS (= no custom keys/values in Firebase console).

But works fine on Android (= the customs keys/values are displayed).

**Comment by kroikie** [comment]
@sperochon thanks for reporting that, I'll try to reproduce on iOS.

**Comment by behnam-lum** [comment]
I was also running an iOS app, it should explain why I have the issue.

**Comment by kroikie** [comment]
@behnam-lum thanks for confirming, looks like an iOS issue.

**Comment by behnam-lum** [comment]
Any progress on this? Can I look into anything related to see if I can come up with a fix? Is this an iOS side Crashlytics plugin/sdk issue?

**Comment by sperochon** [comment]
Apologize on our side, it appears that it now works fine on iOS. The problem was that we used an old version of the Crashlytics package. We were not aware of the special meaning of version < 1.0.0 in pubspec.yaml and thought that using ^ would update minor versions.

So, we have updated the version number manually in the pubspec.yaml to the latest:
firebase_crashlytics: ^0.1.0+3

Here is the info: https://stackoverflow.com/a/53563563/5553628

Sorry again for the false alert.
